frontend/packages/metal3-plugin/src/components/dashboard/events-card.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventModel, MachineModel } from '@console/internal/models';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { EventsBody } from '@console/internal/components/dashboard/events-card/events-body';
+import { EventKind, K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { DashboardItemProps } from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { getName, getNamespace } from '@console/shared';
+
+import { getHostMachineName } from '../../selectors';
+import { BaremetalHostModel } from '../../models';
+
+const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+const getMachineResource = (name: string, namespace: string): FirehoseResource => ({
+  isList: false,
+  namespace,
+  name,
+  kind: referenceForModel(MachineModel),
+  prop: 'machine',
+});
+
+const matchesHost = (machine: K8sResourceKind, event: EventKind): boolean =>
+  machine && _.get(event, 'source.host') === _.get(machine, 'status.nodeRef.name');
```

Comment:
you can use `getMachineNodeName` from `@console/shared` instead of `_.get(machine, 'status.nodeRef.name');`

Reply:
good to know, using `getMachineNodeName` now